File: dropwizard-extra-common/src/main/scala/com/datasift/dropwizard/config/GraphiteReportingConfiguration.scala
package com.datasift.dropwizard.config

import com.yammer.dropwizard.config.Configuration
import reflect.BeanProperty
import javax.validation.Valid
import javax.validation.constraints.NotNull

/** [[com.yammer.dropwizard.config.Configuration]] mix-in for Graphite support */
trait GraphiteReportingConfiguration {
  self: Configuration =>

  /** configuration for Graphite metrics reporting */
  @BeanProperty
  @NotNull
  @Valid
  val graphite: GraphiteConfiguration = new GraphiteConfiguration
}

File: dropwizard-extra-core/src/main/java/com/datasift/dropwizard/config/GraphiteReportingConfiguration.java
package com.datasift.dropwizard.config;

import com.codahale.dropwizard.Configuration;

/**
 * Interface for {@link Configuration}s that include Graphite metrics reporting capability.
 */
public interface GraphiteReportingConfiguration {

    /**
     * Gets the configuration of the Graphite instance to report metrics to.
     *
     * @return the configuration of the Graphite instance to report metrics to.
     */
    public GraphiteConfiguration getGraphite();
}
